kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/le/beacon/AdvertisingReportAddressType.java

```
@@ -0,0 +1,31 @@
+package org.eclipse.kura.bluetooth.le.beacon;
+
+public enum AdvertisingReportAddressType {
+
+    PUBLIC((byte) 0x00), // Public Device Address
+    RANDOM((byte) 0x01), // Random Device Address
+    RESERVED((byte) 0xFF); // Reserved for future use
+
+    private final byte addressType;
+
+    private AdvertisingReportAddressType(byte addressType) {
+        this.addressType = addressType;
+    }
+
+    public byte getEventTypeCode() {
+        return this.addressType;
+    }
+
+    public static AdvertisingReportAddressType getAddressType(byte address) {
+        AdvertisingReportAddressType type = null;
```

to be consistent with other comments in this review, instead of returning null, wouldn't be better to return a special UNKNOWN value or throw an exception?

---

We return RESERVED.